application/src/bll/reducers/cartReducer.test.tsx

```diff
@@ -0,0 +1,37 @@
+import {
+  cartReducer,
+  addToCart,
+  addAmount,
+  deleteCar,
+  cleanCart,
+  CartPropTypes
+} from './cartReducer';
+import cars from '../../data/database.json';
+
+describe('Cart reducer tests', () => {
+  let initState: Array<CartPropTypes> = [];
+  const defaultCartState: Array<CartPropTypes> = [];
+  let newState: Array<CartPropTypes> = [];
+  beforeEach(() => {
+    initState = cartReducer(defaultCartState, addToCart(cars[0]));
+  });
+  test('add car to cart', () => {
+    expect(initState).toHaveLength(1);
+  });
+  test('add amount to cars', () => {
+    newState = cartReducer(initState, addAmount(cars[0].id, 2));
+    expect(newState[0].id).toBe(cars[0].id);
+    expect(newState[0].amount).toBeGreaterThan(1);
+  });
+  test('delete car', () => {
+    newState = cartReducer(initState, deleteCar(cars[0].id));
+    expect(newState.length).toBeLessThan(2);
```

Comment:
2 is magic number

`const initialCartItemsAmount = 3`;
`const updatedCartItemsAmount = 2`;

So you should check length of items before&after deleting. 
```
```

Reply:
i fixed everywhere